tests/posix/threads_base/prj.conf

```
@@ -0,0 +1,15 @@
+CONFIG_POSIX_API=y
+CONFIG_ZTEST=y
+
+CONFIG_POSIX_AEP_CHOICE_BASE=y
+CONFIG_POSIX_THREADS=y
+CONFIG_POSIX_PRIORITY_SCHEDULING=y
+CONFIG_POSIX_THREAD_PRIORITY_SCHEDULING=y
+CONFIG_THREAD_NAME=y
+CONFIG_DYNAMIC_THREAD=y
+CONFIG_THREAD_STACK_INFO=y
+CONFIG_DYNAMIC_THREAD_POOL_SIZE=6
+CONFIG_POSIX_THREAD_THREADS_MAX=6
+CONFIG_HEAP_MEM_POOL_SIZE=4096
+CONFIG_POSIX_THREAD_KEYS_MAX=512
```

Given that `NR_KEY` is set to 2 in the testsuite, we might be able to get away with setting these values to 6 (2 keys for each of the 3 possible threads)
```suggestion
CONFIG_POSIX_THREAD_KEYS_MAX=6
```